src/main/java/org/ibp/api/rest/audit/GermplasmAuditResource.java

```diff
@@ -0,0 +1,182 @@
+
+	@ResponseBody
+	@RequestMapping(
+		value = "/crops/{cropName}/germplasm/{gid}/basic-details/changes",
+		method = RequestMethod.GET)
+	@PreAuthorize(HAS_VIEW_CHANGE_HISTORY)
+	public ResponseEntity<List<GermplasmBasicDetailsAuditDTO>> getBasicDetailsChanges(
+		@PathVariable final String cropName,
+		@PathVariable final Integer gid,
+		final Pageable pageable) {
+
+		final BindingResult errors = new MapBindingResult(new HashMap<>(), String.class.getName());
+		this.germplasmValidator.validateGermplasmId(errors, gid);
+
+		return this.getPagedResult(() -> this.auditService.countBasicDetailsChangesByGid(gid),
+			() -> this.auditService.getBasicDetailsChangesByGid(gid, pageable),
+			pageable);
+	}
+
+	@ResponseBody
+	@RequestMapping(
+		value = "/crops/{cropName}/germplasm/{gid}/reference/changes",
+		method = RequestMethod.GET)
+	@PreAuthorize(HAS_VIEW_CHANGE_HISTORY)
+	public ResponseEntity<List<GermplasmReferenceAuditDTO>> getReferenceChanges(
+		@PathVariable final String cropName,
+		@PathVariable final Integer gid,
+		final Pageable pageable) {
+
+		final BindingResult errors = new MapBindingResult(new HashMap<>(), String.class.getName());
+		this.germplasmValidator.validateGermplasmId(errors, gid);
+
+		return this.getPagedResult(() -> this.auditService.countReferenceChangesByGid(gid),
+			() -> this.auditService.getReferenceChangesByGid(gid, pageable),
+			pageable);
+	}
+
+	@ResponseBody
+	@RequestMapping(
+		value = "/crops/{cropName}/germplasm/{gid}/progenitor-details/changes",
+		method = RequestMethod.GET)
+	@PreAuthorize(HAS_VIEW_CHANGE_HISTORY)
+	public ResponseEntity<List<GermplasmProgenitorDetailsAuditDTO>> getProgenitorDetailsChanges(
+		@PathVariable final String cropName,
+		@PathVariable final Integer gid,
+		final Pageable pageable) {
+
+		final BindingResult errors = new MapBindingResult(new HashMap<>(), String.class.getName());
+		this.germplasmValidator.validateGermplasmId(errors, gid);
+
+		return this.getPagedResult(() -> this.auditService.countProgenitorDetailsChangesByGid(gid),
+			() -> this.auditService.getProgenitorDetailsChangesByGid(gid, pageable),
+			pageable);
+	}
+
+	@ResponseBody
+	@RequestMapping(
+		value = "/crops/{cropName}/germplasm/{gid}/progenitors-other/changes",
+		method = RequestMethod.GET)
+	@PreAuthorize(HAS_VIEW_CHANGE_HISTORY)
+	public ResponseEntity<List<GermplasmOtherProgenitorsAuditDTO>> getOtherProgenitorsChanges(
+		@PathVariable final String cropName,
+		@PathVariable final Integer gid,
+		final Pageable pageable) {
+
+		final BindingResult errors = new MapBindingResult(new HashMap<>(), String.class.getName());
+		this.germplasmValidator.validateGermplasmId(errors, gid);
+
+		return this.getPagedResult(() -> this.auditService.countOtherProgenitorsChangesByGid(gid),
+			() -> this.auditService.getOtherProgenitorsByGid(gid, pageable),
+			pageable);
+	}
+
+	private <T> ResponseEntity<List<T>> getPagedResult(final Supplier<Long> countSupplier, final Supplier<List<T>> resultsSupplier,
```

Comment:
Seems this can be moved to an util class? 
maybe we can start implementing this for new and existing services? 
@lik-83